akka-actor/src/main/scala/akka/actor/ActorRef.scala

```diff
@@ -599,6 +599,31 @@ private[akka] class VirtualPathContainer(
 
   private val children = new ConcurrentHashMap[String, InternalActorRef]
 
+  /**
+   * In [[ActorSelectionMessage]]s only [[SelectChildName]] elements
+   * are supported, otherwise messages are sent to [[EmptyLocalActorRef]].
+   */
+  override def !(message: Any)(implicit sender: ActorRef = Actor.noSender): Unit = message match {
+    case sel @ ActorSelectionMessage(msg, elements, wildcardFanOut) ⇒ {
+      require(elements.nonEmpty)
+
+      elements.head match {
+        case SelectChildName(name) ⇒
+          val child = getChild(name)
+          if (elements.tail.isEmpty) {
+            child ! msg
+          } else {
+            child ! sel.copy(elements = elements.tail)
```

Comment:
we can't support this case either, because this will complete the Future with the remaining ActorSelection message, i.e.  EmptyLocalActorRef here also

test case: `context.actorSelection("/temp/" + name + "/missing") ! x`